test/fixtures/dialects/tsql/nested_joins.sql

```diff
@@ -18,3 +18,11 @@ RIGHT OUTER JOIN I
             ON P_2.NID = IP_2.NID
         ON I.SecondaryPID = CAST(P_2.IDEID AS varchar)
     ON CAST(BA.IDEID AS varchar) = I.BAID
+
+SELECT 1 AS RegionCode
+FROM BA
+LEFT OUTER JOIN (
+    I JOIN P
+        ON I.Pcd = P.Iid
+) ON BA.Iid = I.Bcd;
+GO
```

Comment:
When we lint this, it causes L026 to incorrectly fire:

```
L:   5 | P:  12 | L026 | Reference 'I.Pcd' refers to table/view not found in the
                       | FROM clause or found in ancestor statement.
L:   5 | P:  20 | L026 | Reference 'P.Iid' refers to table/view not found in the
                       | FROM clause or found in ancestor statement.
L:   6 | P:  15 | L026 | Reference 'I.Bcd' refers to table/view not found in the
                       | FROM clause or found in ancestor statement.
```

Think we need to see if we can tweak L026 to account for this before we can merge

Reply:
Think I've found the fix for this so pushing to this branch if you don't mind.

Reply:
Excellent. I tried to do something similar but your code is definitely nicer.